contracts/lib/zeppelin/math/SafeMath.sol

```diff
@@ -1,32 +1,69 @@
-pragma solidity ^0.4.11;
+// See https://github.com/OpenZeppelin/openzeppelin-solidity/blob/d51e38758e1d985661534534d5c61e27bece5042/contracts/math/SafeMath.sol
+// Adapted to use pragma ^0.4.18 and satisfy our linter rules
+
+pragma solidity ^0.4.18;
 
 
 /**
  * @title SafeMath
- * @dev Math operations with safety checks that throw on error
+ * @dev Math operations with safety checks that revert on error
  */
 library SafeMath {
-  function mul(uint256 a, uint256 b) internal pure returns (uint256) {
-    uint256 c = a * b;
-    require(a == 0 || c / a == b);
-    return c;
-  }
-
-  function div(uint256 a, uint256 b) internal pure returns (uint256) {
-    // assert(b > 0); // Solidity automatically throws when dividing by 0
-    uint256 c = a / b;
-    // assert(a == b * c + a % b); // There is no case in which this doesn't hold
-    return c;
-  }
-
-  function sub(uint256 a, uint256 b) internal pure returns (uint256) {
-    require(b <= a);
-    return a - b;
-  }
-
-  function add(uint256 a, uint256 b) internal pure returns (uint256) {
-    uint256 c = a + b;
-    require(c >= a);
-    return c;
-  }
+
+    /**
+    * @dev Multiplies two numbers, reverts on overflow.
+    */
+    function mul(uint256 _a, uint256 _b) internal pure returns (uint256) {
+        // Gas optimization: this is cheaper than requiring 'a' not being zero, but the
+        // benefit is lost if 'b' is also tested.
+        // See: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/522
+        if (_a == 0) {
+            return 0;
+        }
+
+        uint256 c = _a * _b;
+        require(c / _a == _b);
```

Comment:
Are we checking anything else than overflow here? If it's just overflow, we could do:
```
        uint256 c = _a * _b;
        if (_a > 1) { 
            if(_b > 1) {
                require(c > _a);
                require(c > _b);
            }
        }
        return c;
```
Which is another small gas gain.

Reply:
Not sure if this is actually a gas gain, but there might also be some numbers `a` and `b` where `(a * b) % uint256_max > max(a, b)`.

Probably best to suggest it to open-zeppelin if it turns out it is a gas gain :).

Reply:
> Not sure if this is actually a gas gain, 

Yes, there is, I did some tests with Remix. Small though.

> but there might also be some numbers a and b where (a * b) % uint256_max > max(a, b).

I thought a little bit about it and I don't think so. I'll try to sketch a mathematical proof to be sure. Have you found a counter-example?

> Probably best to suggest it to open-zeppelin if it turns out it is a gas gain :).

You are absolutely right, I'll bring it up there. (Unless you found that counter-example, of course)

Reply:
Haven't, but a small script will probably find one or exhaust all cases pretty quickly, at least for `uint256_max`. Given arbitrary `a`, `b`, and constant `C` though, intuitively there should be counter examples (`uint256_max` might not as a special case).

An example would be `a = 3`, `b = 3`, `C = 5`; `(3 * 3) % 5` is `4`, which satisfies the above.

Reply:
Oh, yes, I was so wrong! (sorry for distracting you). A similar counter-example would work for uint256: (2^255+1)*3. So forget it (and sorry again, my intuition was completely wrong and I should have checked it correctly before).